core/domain/suggestion_domain.py

```diff
@@ -72,3 +77,73 @@ def to_dict(self):
             'change_cmd': self.change_cmd,
             'score_category': self.score_category
         }
+
+    def validate(self):
+        """Validates the suggestion object. Each subclass must implement
+        this function
+        """
+        pass
+
+
+class SuggestionEditStateContent(BaseSuggestion):
+    """Domain object for a suggestion of type SUGGESTION_EDIT_STATE_CONTENT.
+
+    Attributes:
```

Comment:
I don't think you need to repeat the attributes here.

Reply:
In that case would we have to rewrite the attributes in all the sub-classes? The set of attributes is the same for all the subclasses. So wouldn't it be better to have it only at the superclass level (though the methods there will not be called).

Reply:
I guess the question is where the logic that uses those attrs (e.g. setters/getters) resides. If in the superclass, then it makes sense to document them there. Otherwise it probably goes in the subclasses.

If you can find a way to move common logic to the superclass, that would be great too.